From 37abd5d34e18d11ff2961f672cf4491471029684 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 28 Feb 2013 23:39:18 -0400
Subject: [PATCH] hacked up to build on Android

removing stuff I don't need and stuff removed from other modules
---
 Yesod.hs    |    7 ------
 yesod.cabal |   77 -----------------------------------------------------------
 2 files changed, 84 deletions(-)

diff --git a/Yesod.hs b/Yesod.hs
index ef9623d..255ab56 100644
--- a/Yesod.hs
+++ b/Yesod.hs
@@ -6,7 +6,6 @@ module Yesod
       module Yesod.Core
     , module Yesod.Form
     , module Yesod.Json
-    , module Yesod.Persist
       -- * Running your application
     , warp
     , warpDebug
@@ -21,19 +20,14 @@ module Yesod
     , readIntegral
       -- * Hamlet library
       -- ** Hamlet
-    , hamlet
-    , xhamlet
     , HtmlUrl
     , Html
     , toHtml
       -- ** Julius
-    , julius
     , JavascriptUrl
     , renderJavascriptUrl
     , toJSON
       -- ** Cassius/Lucius
-    , cassius
-    , lucius
     , CssUrl
     , renderCssUrl
     ) where
@@ -46,7 +40,6 @@ import Text.Julius
 
 import Yesod.Form
 import Yesod.Json
-import Yesod.Persist
 import Control.Monad.IO.Class (liftIO, MonadIO(..))
 import Control.Monad.Trans.Control (MonadBaseControl)
 
diff --git a/yesod.cabal b/yesod.cabal
index 741f19a..7566cfb 100644
--- a/yesod.cabal
+++ b/yesod.cabal
@@ -13,7 +13,6 @@ description:
     The Yesod documentation site <http://www.yesodweb.com/> has much more information, including on the supporting packages mentioned above.
 category:        Web, Yesod
 stability:       Stable
-cabal-version:   >= 1.6
 build-type:      Simple
 homepage:        http://www.yesodweb.com/
 
@@ -28,9 +27,7 @@ extra-source-files:
 library
     build-depends:   base                      >= 4.3      && < 5
                    , yesod-core                >= 1.1.5    && < 1.2
-                   , yesod-auth                >= 1.1      && < 1.2
                    , yesod-json                >= 1.1      && < 1.2
-                   , yesod-persistent          >= 1.1      && < 1.2
                    , yesod-form                >= 1.1      && < 1.3
                    , yesod-default             >= 1.1.3    && < 1.2
                    , monad-control             >= 0.3      && < 0.4
@@ -48,80 +45,6 @@ library
     exposed-modules: Yesod
     ghc-options:     -Wall
 
-executable             yesod-ghc-wrapper
-    main-is: ghcwrapper.hs
-    build-depends:
-                    base                       >= 4         && < 5
-                  , Cabal
-
-executable             yesod-ld-wrapper
-    main-is: ghcwrapper.hs
-    cpp-options:     -DLDCMD
-    build-depends:
-                    base                       >= 4         && < 5
-                  , Cabal
-executable             yesod-ar-wrapper
-    main-is: ghcwrapper.hs
-    cpp-options:     -DARCMD
-    build-depends:
-                     base                       >= 4         && < 5
-                   , Cabal
-
-executable             yesod
-    if os(windows)
-        cpp-options:     -DWINDOWS
-    build-depends:     base               >= 4.3          && < 5
-                     , ghc                >= 7.0.3        && < 7.8
-                     , ghc-paths          >= 0.1
-                     , parsec             >= 2.1          && < 4
-                     , text               >= 0.11
-                     , shakespeare-text   >= 1.0          && < 1.1
-                     , shakespeare        >= 1.0.2        && < 1.1
-                     , shakespeare-js     >= 1.0.2        && < 1.2
-                     , shakespeare-css    >= 1.0.2        && < 1.1
-                     , bytestring         >= 0.9.1.4
-                     , time               >= 1.1.4
-                     , template-haskell
-                     , directory          >= 1.0
-                     , Cabal
-                     , unix-compat        >= 0.2          && < 0.5
-                     , containers         >= 0.2
-                     , attoparsec         >= 0.10
-                     , http-types         >= 0.7
-                     , blaze-builder      >= 0.2.1.4      && < 0.4
-                     , filepath           >= 1.1
-                     , process
-                     , zlib               >= 0.5          && < 0.6
-                     , tar                >= 0.4          && < 0.5
-                     , system-filepath    >= 0.4          && < 0.5
-                     , system-fileio      >= 0.3          && < 0.4
-                     , unordered-containers
-                     , yaml               >= 0.8          && < 0.9
-                     , optparse-applicative >= 0.4
-                     , fsnotify           >= 0.0          && < 0.1
-                     , split              >= 0.2          && < 0.3
-                     , file-embed
-                     , conduit            >= 0.5          && < 0.6
-                     , resourcet          >= 0.3          && < 0.5
-                     , base64-bytestring
-                     , lifted-base
-                     , http-reverse-proxy >= 0.1.1
-                     , network
-                     , http-conduit
-                     , network-conduit
-                     , project-template   >= 0.1.1
-
-    ghc-options:       -Wall -threaded
-    main-is:           main.hs
-    other-modules:     Scaffolding.Scaffolder
-                       Devel
-                       Build
-                       GhcBuild
-                       Keter
-                       AddHandler
-                       Paths_yesod
-                       Options
-
 source-repository head
   type:     git
   location: https://github.com/yesodweb/yesod
-- 
1.7.10.4

